### PR TITLE
Add OpenSearch features.

### LINKED ISSUE #2414: Support OpenSearch in starter
https://micronaut-projects.github.io/micronaut-opensearch/snapshot/guide/

I think we will need three features: 

- [opensearch-amazon](https://micronaut-projects.github.io/micronaut-opensearch/snapshot/guide/#openSearchAmazon)
- [opensearch-restclient](https://micronaut-projects.github.io/micronaut-opensearch/snapshot/guide/#openSearchRestClient)
- [opensearch-httpclient5](https://micronaut-projects.github.io/micronaut-opensearch/snapshot/guide/#openSearchHttpClient5)

### PR DESCRIPTION
**Pull request by wetted**
closes #2414

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/opensearch/OpenSearchFeature.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    default String getThirdPartyDocumentation() {
+        return "https://opensearch.org/docs/latest/clients/java/";
+    }
+
+    @Override
+    default String getCategory() {
+        return Category.CLIENT;
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/opensearch/OpenSearchFeature.java]: category should be `SEARCH_ENGINE` instead. the same elastic search feature uses. 

https://github.com/micronaut-projects/micronaut-starter/blob/4.4.x/starter-core/src/main/java/io/micronaut/starter/feature/Category.java#L49

File: starter-core/src/main/java/io/micronaut/starter/feature/opensearch/OpenSearchRestClient.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import io.micronaut.starter.feature.json.JacksonDatabindFeature;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class OpenSearchRestClient implements OpenSearchFeature {
+
+    public static final String NAME = "opensearch-restclient";
+
+    private final JacksonDatabindFeature jacksonDatabindFeature;
+    private final Dependency jacksonDatabindDependency = MicronautDependencyUtils
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/opensearch/OpenSearchRestClient.java]: remove unused variable.

File: starter-core/src/test/groovy/io/micronaut/starter/feature/opensearch/OpenSearchFeatureSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            assert verifier.hasDependency("io.micronaut", "micronaut-jackson-databind")
+        }
+
+        where:
+        [opensearchFeature, buildTool] << [beanContext.getBeansOfType(OpenSearchFeature), BuildTool.values()].combinations()
+    }
+
+    void "test opensearch feature #opensearchFeature.name is client category"(OpenSearchFeature opensearchFeature) {
+        expect:
+        Category.CLIENT == opensearchFeature.getCategory()
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/opensearch/OpenSearchFeatureSpec.groovy]: this should be SEARCH_ENGINE

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]
if the user selects Testcontainers. We should add https://github.com/micronaut-projects/micronaut-opensearch/blob/master/gradle/libs.versions.toml#L44 see https://github.com/micronaut-projects/micronaut-starter/pull/2413

**Comment by wetted** [comment]
> if the user selects Testcontainers. We should add https://github.com/micronaut-projects/micronaut-opensearch/blob/master/gradle/libs.versions.toml#L44 see #2413

I'll wait until #2413 is merged and make this a `ContributingTestContainerArtifactId` feature.

**Comment by sdelamo** [comment]
@wetted #2413 is merged.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**